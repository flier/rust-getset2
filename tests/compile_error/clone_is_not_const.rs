use getset2::Getter;

#[derive(Default, Getter)]
#[get(pub, const, mut)]
struct Foo {
    #[get(clone)]
    clone_field: usize,
}

fn main() {}
