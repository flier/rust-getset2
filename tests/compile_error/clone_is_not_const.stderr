error[E0015]: cannot call non-const fn `<usize as Clone>::clone` in constant functions
 --> tests/compile_error/clone_is_not_const.rs:6:5
  |
6 |     #[get(clone)]
  |     ^
  |
  = note: calls in constant functions are limited to constant functions, tuple structs and tuple variants
